From ac3797dfc1a0290cedfb35a883af1d171ab22be0 Mon Sep 17 00:00:00 2001
From: Benjamin Otte <otte@redhat.com>
Date: Wed, 16 Jun 2010 10:41:40 +0000
Subject: Can't select file on file browser after changing sort order

The row values are 1-indexed not 0-indexed, this has to be taken into
account when producing the new_order array.

https://bugzilla.gnome.org/show_bug.cgi?id=621414
(cherry picked from commit a8e3ce6bb486a4c89799a77abe153d647fea1a69)
---
diff --git a/gtk/gtkfilesystemmodel.c b/gtk/gtkfilesystemmodel.c
index 738b061..7c9e1dc 100644
--- a/gtk/gtkfilesystemmodel.c
+++ b/gtk/gtkfilesystemmodel.c
@@ -739,7 +739,7 @@ gtk_file_system_model_sort (GtkFileSystemModel *model)
                   continue;
                 }
 
-              new_order[r] = node->row;
+              new_order[r] = node->row - 1;
               r++;
               node->row = r;
             }
--
cgit v0.8.3.1
